src/clj/rems/application/commands.clj

```diff
@@ -299,13 +297,14 @@
                       :event/attachments (when-let [att (:attachments cmd)]
                                            (vec att))))))
 
-(defn- build-forms-list [catalogue-item-ids {:keys [get-catalogue-item]}]
+(defn- build-forms-list [workflow catalogue-item-ids {:keys [get-catalogue-item]}]
   (->> catalogue-item-ids
        (mapv get-catalogue-item)
        (mapv :formid)
        (remove nil?)
-       (distinct)
-       (mapv (fn [form-id] {:form/id form-id}))))
+       (mapv (fn [form-id] {:form/id form-id}))
+       (concat (get-in workflow [:workflow :forms])) ; NB: workflow forms end up first
```

Comment:
is ordering significant for these forms or could we sort by id?

Reply:
There is no definite rule. The workflow forms were originally put as first, as I think they are likely to contain "common stuff". Otherwise the order of the forms should be kept stable, so one can control the order in which they appear. The order of the catalogue items / resources determines the form order mostly.

Ordering by form title or other condition could be sensible too.